libraries/chain/market_evaluator.cpp

```diff
@@ -275,4 +275,61 @@ void_result call_order_update_evaluator::do_apply(const call_order_update_operat
    return void_result();
 } FC_CAPTURE_AND_RETHROW( (o) ) }
 
+void_result bid_collateral_evaluator::do_evaluate(const bid_collateral_operation& o)
+{ try {
+   database& d = db();
+
+   FC_ASSERT( d.head_block_time() >= HARDFORK_CORE_216_TIME, "Not yet!" );
```

Comment:
Will fix >.

I don't think white/blacklisting are relevant here, because you can only bid BACK that you already have, and you'll only ever get BACK out again (possibly more than you put in). You can't transfer or trade BACK or SWAN here, and you cannot get SWAN out.